api/src/services/platform-service.ts

```diff
@@ -196,4 +244,151 @@ export class PlatformService extends DBService {
+      const artifactZip = new AdmZip();
+      artifactZip.addFile(attachment.file_name, s3File.Body as Buffer);
+
+      return {
+        dataPackageId,
+        archiveFile: {
+          data: artifactZip.toBuffer(),
+          fileName: `${attachment.uuid}.zip`,
+          mimeType: 'application/zip'
+        },
+        metadata: {
+          file_name: attachment.file_name,
+          file_size: attachment.file_size,
+          file_type: isReportAttachment(attachment) ? 'Report' : attachment.file_type || 'Other',
+          title: attachment.title,
+          description: attachment.description
+        }
+      };
+    });
+
+    return Promise.all(promises);
+  }
+
+  /**
+   * Makes a request to the BioHub API to submit an artifact.
+   *
+   * @param {IArtifact} artifact The artifact to submit to BioHub
+   * @returns {*} {Promise<{artifact_id: number>}} The ID belonging to the artifact record in BioHub
+   *
+   * @memberof PlatformService
+   */
+  async _submitArtifactToBioHub(artifact: IArtifact): Promise<{ artifact_id: number }> {
+    const keycloakService = new KeycloakService();
+
+    const token = await keycloakService.getKeycloakToken();
+
+    const formData = new FormData();
+
+    formData.append('media', artifact.archiveFile.data, {
+      filename: artifact.archiveFile.fileName,
+      contentType: artifact.archiveFile.mimeType
+    });
+
+    formData.append('data_package_id', artifact.dataPackageId);
+
+    Object.entries(artifact.metadata).forEach(([metadataKey, metadataValue]) => {
+      formData.append(`metadata[${metadataKey}]`, metadataValue);
+    });
+
+    const backboneArtifactIntakeUrl = new URL(this.backboneArtifactIntakePath, this.backboneApiHost).href;
+
+    const { data } = await axios.post<{ artifact_id: number }>(backboneArtifactIntakeUrl, formData.getBuffer(), {
+      headers: {
+        authorization: `Bearer ${token}`,
+        ...formData.getHeaders()
+      }
+    });
+
+    return data;
+  }
+
+  /**
+   * Uploads the given project attachments and report attachments to BioHub.
+   *
+   * @param {string} dataPackageId The dataPackageId for the artifact submission
+   * @param {number} projectId The ID of the project
+   * @param {number[]} attachmentIds The particular IDs of the attachments to submit to BioHub
+   * @param {number[]} reportAttachmentIds The particular IDs of the report attachments to submit to BioHub
+   * @returns {*} {Promise<{artifact_id: number}[]>} The IDs of all the artifact records in BioHub
+   *
+   * @memberof PlatformService
+   */
+  async uploadProjectAttachmentsToBioHub(
+    dataPackageId: string,
+    projectId: number,
+    attachmentIds: number[],
+    reportAttachmentIds: number[]
+  ): Promise<{ artifact_id: number }[]> {
+    const attachments = (
+      await this.attachmentService.getProjectAttachments(projectId)
+    ).filter((attachment: IProjectAttachment) => attachmentIds.includes(attachment.id));
```

Comment:
You could consider making a new function that gets attachments based on an array of ids (select * where in) type call. Then you wouldn't have to filter them here.